### PR TITLE
Add Kinesis endpoint to NewStream to allow for localtesting against localstack

### PR DESCRIPTION
**Pull request by aaron-collier**
A small suggestion that allows this code to be used against a [localstack](https://github.com/localstack/localstack) instance for local testing/dev.

A bit more work should probably go into this to test and assume a default.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ func main() {
 		Interval:     time.Millisecond * 1000,
 		IteratorType: kcl.IteratorTypeLatest,
 	}
-	k, err := kcl.NewStream(sess, os.Getenv("AWS_KINESIS_STREAM"), s, config)
+	k, err := kcl.NewStream(sess, os.Getenv("AWS_KINESIS_ENDPIONT"), os.Getenv("AWS_KINESIS_STREAM"), s, config)
```
Comment (huttotw) [inline README.md]: Thanks for your contribution, will definitely merge, but can we fix the slight typo?

**Comment by aaron-collier** [comment]
@huttotw fixed. Thanks for catching that :-)

**Comment by huttotw** [comment]
Thanks!

**Comment by jaredpiedt** [comment]
@aaron-collier were you unable to specify the endpoint in the session and then pass that to `NewStream`?
```go
sess, err := session.NewSession(&aws.Config{
	Endpoint: aws.String(os.Getenv("AWS_KINESIS_ENDPOINT")),
})
if err != nil {
	panic(err)
}

s := kcl.NewLocalStore()
config := kcl.Config{
	Limit:        1000,
	Interval:     time.Millisecond * 1000,
	IteratorType: kcl.IteratorTypeLatest,
}
k, err := kcl.NewStream(sess, os.Getenv("AWS_KINESIS_STREAM"), s, config)
if err != nil {
	panic(err)
}
```

**Comment by aaron-collier** [comment]
@jaredpiedt Good Question. It's been awhile, but in this case the app uses several various AWS endpoints and our devops strategy here was to limit the number of sessions created (off hand, I can't recall the specifics of why), so I don't think setting it on the session would have worked out... But I'll have to investigate this when we get back to the project. Thanks for the insight.

**Pull request merged**